op-node/rollup/driver/sync_deriver.go

```diff
@@ -232,6 +232,14 @@ func (s *SyncDeriver) SyncStep() {
 		return
 	}
 
+	if s.SyncCfg.SafeSource == sync.SafeSourceL2 {
+		// The pipeline cannot move forwards when using L2 safe source (derivation is skipped).
+		s.Log.Debug("Rollup driver is backing off because using L2 safe source.",
+			"unsafe_head", s.Engine.UnsafeL2Head())
+		s.StepDeriver.ResetStepBackoff(s.Ctx)
```

Comment:
I believe that in traditional EL sync we need to `ResetStepBackoff` because both safe and unsafe block progression is paused while the EL is catching up -- tell me if I'm incorrect there.

If I am correct, then we should actually **not** `ResetStepBackoff` here and instead simply return early because when SafeSource=L2 we will still want to be progressing the unsafe head at max speed. Also it's worth noting that in the case where we are doing an EL sync (as a part of normal SafeSource=L2 functioning) we will still get to make use of the original `ResetStepBackoff` in the above if statement `if s.Engine.IsEngineSyncing()`.

Am I correct here in what ResetStepBackoff does?